src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -371,7 +371,10 @@ private Screen autoAdvanceSession(
             previousScreen = nextScreen;
             iterationCount += 1;
 
-            nextScreen = menuSession.getNextScreen(needsFullEntityScreen, entityScreenContext);
+            // skips the call to to getNextScreen for first iteration if we already have nextScreen
+            if (iterationCount > 1 || nextScreen == null) {
+                nextScreen = menuSession.getNextScreen(needsFullEntityScreen, entityScreenContext);
+            }
```

Comment:
This is the crux of the issue where we are no longer calling getNextScreen for the first iteration if we already have it evaluated.